client/src/components/Login/SignUpForm/index.js

```diff
@@ -59,11 +61,60 @@ const SignUpForm = ({ switchView }) => {
     }
   };
 
+  const validateUsername = () => {
+    let usernameState = {
+      error: false,
+      message: ""
+    };
+    if (!username) {
+      usernameState.error = true;
+      usernameState.message = "Username is required";
+    }
+    setValidationState({
+      ...validationState,
+      username: usernameState
+    });
+  };
+
+  const validatePassword = () => {
+    let passwordState = {
+      error: false,
+      message: ""
+    };
+    if (!password) {
+      passwordState.error = true;
+      passwordState.message = "Password is required.";
+    } else if (password.length < 8) {
+      passwordState.error = true;
+      passwordState.message = "Password must be at least 8 characters.";
+    }
+    setValidationState({
+      ...validationState,
+      password: passwordState
+    });
+  };
+
+  const signUp = async () => {
+    try {
+        const signUpStatus = await ServiceClient.signUp({ name, username, password });
+        if (signUpStatus.success) {
+            updateAuthToken && udpateAuthtoken(signUpStatus.authToken);
+            redirectToHome(); // ?
+        }
+        else if (signUpStatus.success ?? false) {
+            setErrorMessage(signUpStatus.message);
+        }
+    } catch (error) {
+        setSignUpError(true);
+    }
+  }
+
   const handleSubmit = (e) => {
     e.preventDefault();
     console.log("clicked sign in");
```

Comment:
You can go ahead and get rid of these console.log statements. In general, you never want to commit a console.log to the master branch.